DFE-Digital/Dfe.Academies.External
# 104118 plumb in api post method

As part of the below ticket, plumb in POST API method ONLY:-
https://dfe-gov-uk.visualstudio.com.mcas.ms/Academies-and-Free-Schools-SIP/_workitems/edit/104118?McasTsid=26110

## Type of change

## Expected behaviour
Now, when the user presses 'save & continue' on what is your role page an application gets created within the database

## Steps to reproduce issue (if relevant)
n/a

## Steps to test this PR
1. Run Unit Tests to ensure code amended not broken. Done by CI / CD
2. Create an application through the application pages by:-
a) logging in
b) start new application
c) selecting JoinAMat
d) selecting what you want within role
e) press save & continue

## Prerequisites
n/a
